collector/queries.go

```diff
@@ -0,0 +1,415 @@
+	`
+
+	// ===== SQL Warehouse Queries =====
+
+	// queriesQuery retrieves total SQL query counts per workspace and warehouse.
+	// Note: query.history can be high volume, limiting to last 1 hour.
+	// Extracts warehouse_id from compute STRUCT.
+	queriesQuery = `
+		SELECT 
+			workspace_id,
+			COALESCE(compute.warehouse_id, 'unknown') as warehouse_id,
+			COUNT(*) as query_count
+		FROM system.query.history
+		WHERE start_time >= current_timestamp() - INTERVAL 1 HOURS
+		GROUP BY workspace_id, compute.warehouse_id
+	`
+
+	// queryErrorsQuery counts failed queries per workspace and warehouse.
+	// A query is considered failed if error_message is not null.
+	// Extracts warehouse_id from compute STRUCT.
+	queryErrorsQuery = `
+		SELECT 
+			workspace_id,
+			COALESCE(compute.warehouse_id, 'unknown') as warehouse_id,
+			COUNT(*) as error_count
+		FROM system.query.history
+		WHERE start_time >= current_timestamp() - INTERVAL 1 HOURS
+			AND error_message IS NOT NULL
+		GROUP BY workspace_id, compute.warehouse_id
+	`
+
+	// queryDurationQuery calculates query latency quantiles per warehouse.
+	// Uses total_duration_ms field from query.history.
+	// Extracts warehouse_id from compute STRUCT.
+	queryDurationQuery = `
+		SELECT 
+			workspace_id,
+			warehouse_id,
+			percentile_approx(duration_seconds, 0.5) as p50,
+			percentile_approx(duration_seconds, 0.95) as p95,
+			percentile_approx(duration_seconds, 0.99) as p99
+		FROM (
+			SELECT 
+				workspace_id,
+				COALESCE(compute.warehouse_id, 'unknown') as warehouse_id,
+				total_duration_ms / 1000.0 as duration_seconds
+			FROM system.query.history
+			WHERE start_time >= current_timestamp() - INTERVAL 1 HOURS
+				AND total_duration_ms IS NOT NULL
+				AND total_duration_ms > 0
+		)
+		GROUP BY workspace_id, warehouse_id
+	`
+
+	// queriesRunningQuery estimates concurrent queries by finding overlapping intervals per warehouse.
+	// This is an approximation - exact concurrency would require real-time polling.
+	// Returns the maximum concurrent queries over the last hour.
+	// Extracts warehouse_id from compute STRUCT.
+	queriesRunningQuery = `
+		SELECT 
+			workspace_id,
+			warehouse_id,
+			MAX(concurrent_count) as max_concurrent
+		FROM (
+			SELECT 
+				q1.workspace_id,
+				COALESCE(q1.compute.warehouse_id, 'unknown') as warehouse_id,
+				q1.start_time as time_point,
+				COUNT(*) as concurrent_count
+			FROM system.query.history q1
+			JOIN system.query.history q2
+				ON q1.workspace_id = q2.workspace_id
+				AND COALESCE(q1.compute.warehouse_id, 'unknown') = COALESCE(q2.compute.warehouse_id, 'unknown')
+				AND q2.start_time <= q1.start_time
+				AND (q2.end_time >= q1.start_time OR q2.end_time IS NULL)
+			WHERE q1.start_time >= current_timestamp() - INTERVAL 1 HOUR
+			GROUP BY q1.workspace_id, q1.compute.warehouse_id, q1.start_time
+		)
+		GROUP BY workspace_id, warehouse_id
+	`
+)
+
+// Time window constants for configurable lookback periods
+const (
```

Comment:
Would be great to use these values directly in the sql queries, eg. the one right above with `INTERVAL 1 HOUR` and another that I commented on with `3600` seconds